### PR TITLE
Improvement: Allow user to manually connect and disconnect

### LINKED ISSUE #41: How to disconnect properly?
Can someone please point me to the proper way to disconnect - I suppose $socket.close() should be used. But if reconnect is enabled - how to disable it. Thank you.

### PR DESCRIPTION
**Pull request by NaughtyMC**
This improvement solves a common use case and resolves #41. By adding a `connectManually` property to the config object, the user can manually connect and disconnect the WebSocket via the `$connect` and `$disconnect` functions. If no property is added the config, functionality is left unchanged (plugin automatically connects when installed).

Please note: this PR might make #46 unnecessary.

### CODE REVIEW

File: src/Main.js [hunk 1]
```diff
@@ -6,9 +6,25 @@ export default {
   install (Vue, connection, opts = {}) {
     if (!connection) { throw new Error('[vue-native-socket] cannot locate connection') }
 
-    let observer = new Observer(connection, opts)
+    let observer = null
 
-    Vue.prototype.$socket = observer.WebSocket
+    if (opts.connectManually) {
+      Vue.prototype.$connect = function () {
```
Comment (weglov) [inline src/Main.js]: Please use arrow functions

**Comment by weglov** [comment]
Tnx for PR, can you bump version to 2.0.7 and update CHANGELOG.md pls?

File: src/Main.js (hunk 1 above)
Comment (NaughtyMC) [reply src/Main.js]: Done

**Comment by NaughtyMC** [comment]
@weglov Sorry I missed that! Thanks for the quick review/approval.

**Comment by danackerson** [comment]
@weglov do you know how long it typically takes for the new release (2.0.7) to make it to the npm repo?

**Comment by weglov** [comment]
@danackerson Not sure about npm @nathantsoi can advise you.
You can already install v2.0.7 from github repo: `npm install https://github.com/nathantsoi/vue-native-websocket.git --save`

**Comment by nathantsoi** [comment]
@weglov, send me your email / npmjs.com username so I can add you to the npm collabs

screen shot 2018-04-09 at 9 27 55 am

same goes for any other contributors who want to help out further!

**Pull request merged**